src/amo/pages/Addon/index.js

```diff
@@ -507,23 +511,32 @@ export class AddonBase extends React.Component {
                   {i18n.gettext('Extension Metadata')}
                 </h2>
               </header>
-              <WrongPlatformWarning
-                addon={addon}
-                className="Addon-WrongPlatformWarning"
-                currentVersion={currentVersion}
-                fixAndroidLinkMessage={i18n.gettext(
-                  `This listing is not intended for this platform.
```

Comment:
I am wondering why we're not encapsulating the Fenix logic inside the `WrongPlatformWarning` component as opposed to passing different props to it?

Reply:
This was done because the message is different on the `Addon` page than it is elsewhere. We could move this logic into the `WrongPlatformWarning`, and just pass a flag into it to say that it's running on the `Addon` page, but it kind of makes sense to allow the page calling it to set the message; it has a better understanding of the context. I believe that I designed it this way also to address the case where there is yet another page that wants the message to be different. That case never emerged, but I'm still happy with the way this is.